src/main/java/org/dita/dost/reader/MergeTopicParser.java

```diff
@@ -202,15 +200,16 @@ public void startDocument() throws SAXException {
     @Override
     public void startElement(final String uri, final String localName, final String qName, final Attributes attributes)
             throws SAXException {
-        // Skip redundant <dita> tags.
-        if (ELEMENT_NAME_DITA.equals(qName)) {
-            rootLang = attributes.getValue(XML_NS_URI, "lang");
-            return;
-        }
         final AttributesImpl atts = new AttributesImpl(attributes);
         final String classValue = atts.getValue(ATTRIBUTE_NAME_CLASS);
 
-        if (TOPIC_TOPIC.matches(classValue)) {
+        if (ELEMENT_NAME_DITA.equals(qName)) {
+            rootLang = attributes.getValue(XML_NS_URI, "lang");
+            if (atts.getValue(ATTRIBUTE_NAME_ID) == null) {
+            	XMLUtils.addOrSetAttribute(atts, ATTRIBUTE_NAME_ID, GENERATED_DITA_ELEMENT_ID);
```

Comment:
It's adding it into `@oid` -- the original ID -- which is not unique. (If every topic in your map has `id="topic"` then every topic in the merged file will have `oid="topic"`.) 

Strictly speaking - it probably shouldn't be necessary. But code today can reasonably assume that any DITA element with `@id` also has `@oid`, so I wanted to avoid any errors from code that tries to use it.

Even then I thought about making them unique by adding a counter, but _if_ somebody is making use of the value, it might actually be better to have the standard token?